json-schema/json-schema-generic/src/test/scala/endpoints/generic/JsonSchemasTest.scala

```diff
@@ -33,6 +33,10 @@ class JsonSchemasTest extends FreeSpec {
       type JsonSchema[+A] = String
       type Record[+A] = String
       type Tagged[+A] = String
+      type Enum[+A] = String
+
+      def enum[A](decode: String => Either[String, A], encode: A => String, values: Seq[A])(implicit tpe: String): String =
+        s"$tpe"
```

Comment:
No. Actually, I don't understand the usefulness of this "fake" interpreter, and had troubles coming up with a string representation. Can you explain?

Now it got me thinking: can we "derive" generic JSON schemas that include enumerations? Not sure how it could work, and there are different ways to represent enums, e.g. as ADT, Scala Enumeration, third-party libs...

Reply:
Oh I missed that this file was used for the “generic” tests only. Its purpose is to easily check what the generically derived schemas look like.

We could implement generic derivation of enumerations later, so we can leave this line as it is.

Reply:
Agreed, I won't implement generic derivation now.